src/main/java/use_case/signup/SignupInteractor.java

```diff
@@ -0,0 +1,40 @@
+package use_case.signup;
+
+import entity.User;
+import use_case.service.UserService;
+
+public class SignupInteractor implements SignupInputBoundary {
+    private final UserService userService;
+    private final SignupOutputBoundary userPresenter;
+//    private final UserFactory userFactory;
+
+    public SignupInteractor(UserService userService,
+                            SignupOutputBoundary signupOutputBoundary) {
+        this.userService = userService;
+        this.userPresenter = signupOutputBoundary;
+//        this.userFactory = userFactory;
+    }
+
+    @Override
+    public void execute(SignupInputData signupInputData) {
+        final String username = signupInputData.getUsername();
+        final String password = signupInputData.getPassword();
+        User user = userService.registerUser(username, password);
```

Comment:
This code will register a user before the check for passwords matching. We can rearrange the conditions in a way so that the password match is checked first